helpers.go

```diff
@@ -0,0 +1,168 @@
+func AppendCommentToObjects(comment string) compiler.Pass {
+	return &compiler.AppendCommentObjects{
+		Comment: comment,
+	}
+}
+
+// PrefixObjectsNames adds the given prefix to every object's name.
+func PrefixObjectsNames(prefix string) compiler.Pass {
+	return &compiler.PrefixObjectNames{
+		Prefix: prefix,
+	}
+}
+
+type CUEOption func(*codegen.CueInput)
+
+// ForceEnvelope decorates the parsed cue Value with an envelope whose
+// name is given. This is useful for dataqueries for example, where the
+// schema doesn't define any suitable top-level object.
+func ForceEnvelope(envelopeName string) CUEOption {
+	return func(input *codegen.CueInput) {
+		input.ForcedEnvelope = envelopeName
+	}
+}
+
+// NameFunc specifies the naming strategy used for objects and references.
+// It is called with the value passed to the top level method or function and
+// the path to the entity being parsed.
+func NameFunc(nameFunc simplecue.NameFunc) CUEOption {
+	return func(input *codegen.CueInput) {
+		input.NameFunc = nameFunc
+	}
+}
+
+// TypesFromSchema generates types from a single input schema and a single
+// output language.
+func TypesFromSchema() *SchemaToTypesPipeline {
+	return &SchemaToTypesPipeline{}
+}
+
+// Debug controls whether debug mode is enabled or not.
+// When enabled, more information is included in the generated output,
+// such as an audit trail of applied transformations.
+func (pipeline *SchemaToTypesPipeline) Debug(enabled bool) *SchemaToTypesPipeline {
+	pipeline.debug = enabled
+	return pipeline
+}
+
+// Run executes the codegen pipeline and returns its output.
+func (pipeline *SchemaToTypesPipeline) Run(ctx context.Context) ([]byte, error) {
+	// Validation
+	if pipeline.input == nil {
+		return nil, fmt.Errorf("no input configured")
+	}
+	if pipeline.output == nil {
+		return nil, fmt.Errorf("no output configured")
+	}
+
+	codegenPipeline, err := codegen.NewPipeline()
+	if err != nil {
+		return nil, err
+	}
+	codegenPipeline.Inputs = []*codegen.Input{pipeline.input}
+	codegenPipeline.Transforms = codegen.Transforms{
+		FinalPasses: pipeline.finalPasses,
+	}
+	codegenPipeline.Output = codegen.Output{
+		Types:     true,
+		Languages: []*codegen.OutputLanguage{pipeline.output},
+	}
+
+	// Run the codegen pipeline and return the generated file's content.
+	// Note: since this pipeline is about types with no runtime, we expect a
+	// single file to be generated.
+	generatedFS, err := codegenPipeline.Run(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	generatedFiles := generatedFS.AsFiles()
+	if len(generatedFiles) != 1 {
+		return nil, fmt.Errorf("expected a single generated file, got %d", len(generatedFiles))
+	}
+
+	return generatedFiles[0].Data, nil
+}
+
+/**********
+ * Inputs *
+ **********/
+
+// CUEValue sets the pipeline's input to the given cue value.
+func (pipeline *SchemaToTypesPipeline) CUEValue(pkgName string, value cue.Value, opts ...CUEOption) *SchemaToTypesPipeline {
```

Comment:
Since this is a builder, I'd expect a `With...` here
```suggestion
func (pipeline *SchemaToTypesPipeline) WithCUEValue(pkgName string, value cue.Value, opts ...CUEOption) *SchemaToTypesPipeline {
```